testing/file_test/testdata/file_only_re_multi_file.carbon

```diff
@@ -0,0 +1,25 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// AUTOUPDATE
+// CHECK:STDOUT: 3 args: `default_args`, `a.carbon`, `b.carbon`
+// CHECK:STDOUT: unattached message 1
+
+// --- a.carbon
+aaa
+// CHECK:STDOUT: file: a.carbon
+// CHECK:STDOUT: unattached message 2
```

Comment:
Why do these messages appear here rather than at the start of the file?

Oh, presumably it's because we only print STDOUT lines immediately after a non-CHECK line, and that works in the main file because we can attach them to the AUTOUPDATE line. Perhaps we need to also print STDOUT CHECK lines at the start of the file in some cases?

Reply:
I tend to agree, but would like to consider it separately since this is how autoupdate is designed to work right now. I'm already asking a structural question on #3214 triggered by STDOUT issues, and #3219 is proposing a significant change for semantics too.